plugins/woocommerce/includes/class-wc-install.php

```diff
@@ -721,16 +741,26 @@ public static function get_db_update_callbacks() {
 
 	/**
 	 * Push all needed DB updates to the queue for processing.
+	 *
+	 * @param bool $schedule_with_delay Whether updates are scheduled to run with a random delay (of up to 15 mins).
 	 */
-	private static function update() {
+	private static function update( bool $schedule_with_delay = false ) {
 		$current_db_version = get_option( 'woocommerce_db_version' );
-		$loop               = 0;
+		$current_wc_version = WC()->version;
 
+		// We add some randomness here prevent overloading servers after an upate has been installed.
+		$scheduled_time = time() + ( $schedule_with_delay ? wp_rand( 1, 15 * MINUTE_IN_SECONDS ) : 0 );
```

Comment:
```suggestion
		$scheduled_time = time() + ( $schedule_with_delay ? wp_rand( 1, 30 * MINUTE_IN_SECONDS ) : 0 );
```
Perhaps I am being over cautious, since this was manual before the gap was like to be spead out in days if not weeks as users come across this update. Should we consider increasing this time bracket to 30 or 60 minutes (or more) to spread out the load a bit more, I am worried that 15 minutes window could be a tight space for the server load to spike, wdyt?

Reply:
Agreed. I changed it to 30 mins and also made the [max delay filterable](https://github.com/woocommerce/woocommerce/pull/54384/commits/39d24b8e0cdcd282ccb8aa582baea240c02bca63#diff-b082a396ea78d8aaa870e896edc388647bd548df1c4eddc9b4953df241faf99bR759-R767), so that it's configurable, probably be hosts. Do you think this is reasonable?

Reply:
>it made sense for this to be in seconds 

I agree; if we keep the `$scheduled_time` in seconds, we are providing a wider range of timestamps for the update to occur. 

> the minimum should be something like 5 minutes and not 0

I am not sure if that would make much difference as we'll have a wide range (with seconds) of randomized timestamps for which the update will be scheduled, but just to be safe, we can keep an initial delay of 2-5 minutes.